main.cpp

```diff
@@ -13,75 +13,505 @@ What you create in this project will be used as the basis of Project 5 in the co
+        //the parameter name is related to the work the function will perform.
+        void fillTank(double fuelAmountInGallons = 2.0);  
+        void breakDown(std::string failureType, bool requiresTow = false);
+        int getMilesTraveledAnnually(bool includeUberLyftTrips);
+    };
+
+    //wash and wax car
+    void washAndWaxCar(Car car); //4) a member function whose parameter is a UDT.
+    //charge customer
+    float chargeCustomer(float discountPercentage);
+    //detail the car interior
+    void detailInterior(Car car);
+    
+    //5) a member variable whose type is a UDT.
+    Car carBeingServiced;  
+};
+
+
+/*
+Thing 1) electric Guitar
+5 properties:
+    1) number of strings (int)
+    2) number of pickups (int)
+    3) color (std::string)
+    4) age (int)
+    5) price (float)
+3 things it can do:
+    1) generate a Note
+    2) break a String
+    3) tune
+ */
+
+struct ElectricGuitar
+{
+    // number of strings (int)
+    int numberOfStrings = 6;
+    // number of pickups (int)
+    int numberOfPickups = 2;
+    // color (std::string)
+    std::string color = "blue";
+    // age (int)
+    int age = 12;
+    // price (float)
+    float price = 1299.99f;
+
+    struct GuitarString
+    {
+        std::string manufacturer = "Ernie Ball";
+        int number = 0;
+        bool isWound = true;
+        float width = 0.52f;
+        std::string material = "steel";
+
+        void breakString();
+        void tune (float pitch);
+        void pluck(int fret); 
+    };
+
+    // generate a Note
+    void generateNote(GuitarString string, int fretPosition);
+    // break a String
+    void breakString(GuitarString stringNumber);
+    // tune
+    void tune(GuitarString stringNumber, float referencePitch); 
+
+    GuitarString string; 
+};
+
+/*
+Thing 2) computer
+5 properties:
+    1) amount of ram (int)
+    2) processor-speed (float)
+    3) size of disk (int)
+    4) manufacturer (std::string)
+    5) operating system (std::string)
+3 things it can do:
+    1) run program
+    2) shut down
+    3) erase disk
+ */
+struct Computer
+{
+    // amount of ram (int) in gigabyte
+    int amountOfRam = 64;
+    // processor-speed (float)in GHz
+    float processorSpeed = 2.4f;
+    // size of disk (int) in terrabyte
+    int sizeOfDisk = 4;
+    // manufacturer (std::string)
+    std::string manufacturer = "Apple";
+    // operating system (std::string)
+    std::string operatingSystem = "MacOS";
+
+    struct Application
+    {
+        std::string name = "Logic Pro";
+        std::string manufacturer = "Apple";
+        std::string appType = "audio";
+        float size = 1.1f;
+        std::string version = "10.5";
+
+        void start();
+        void close();
+        void install(std::string filePath);
+    };
+
+    // run program
+    void runProgram(Application program, int priority);
+    // shut down
+    void shutDown();
+    // erase disk
+    void eraseDisk(std::string volumeName);
+
+    Application application;
```

Comment:
This is a common issue with new programmers or programmers who are coming from garbage-collected languages (python, C#).
When you pass this type into this function, the instance that is used inside this function is NOT the same instance that was passed in.
That means any changes you make to the instance being used INSIDE the function are NOT kept when this function ends.
the C++ concept of References and Pointers has not been covered at this point in the course.  If you wanted this to work as you are most likely expecting it to work, you would need to use references or pointers.
I don't let people use coding concepts in projects until after they've been covered in the course.
You don't need to revise these function parameters.
I just want you to be aware of the potential issue.